posts/a-beginners-guide-to-influxdb.md

```diff
@@ -0,0 +1,109 @@
+---
+authors:
+- Ketki Deshpande
+tags:
+- Data
+date: 
+title: "A Beginner’s Guide to InfluxDB - A Time-Series Database"
+image: 
+---
+
+A time series database (TSDB) is specifically made for data that can be evaluated as a "time-series," which includes sensor outputs, market trends, CPU utilization, and application performance metrics. Most popular types of databases (both SQL and NoSQL) are *not* optimized for handling large-scale scans, or summarization or aggregation of time-series data.Time-series databases have built-in functionality to work with aggregation, down-sampling, data lifecycle management, summarization, etc. Let’s look at one of the popular time-series databases, InfluxDB. InfluxDB uses line protocol for sending and storing data.
+
+# Line Protocol
+The line protocol is the text-based format that uses the syntax below to write points to InfluxDB.
+
+``` <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>] ```
+
+The above syntax is used to insert a single point to an InfluxDB database. Since InfluxDB is a time-series database, the **point** here is analogous to a point on a graph that is representing a time-series data. The collection of points is called a **series**. Next, let’s look at all the four components of the point.
+
+The first component, **Measurement** for an InfluxDB, is similar to a Table for a SQL database. Measurement holds all the tags, fields and time values. It makes sense to name the measurement such that it describes the data contained in it.
+
+The two components **tag set** and **field set** in the line protocol are sets, meaning a key-value pair. **Tag Key** and **Tag Value** are both stored in InfluxDB as strings. In contrast, only **Field Key** can be a string but the **Field Value** can contain strings, integers, floats and booleans too.  The field value is the actual data that we want to save to a database and it is always accompanied by a timestamp, which makes sense since we are inserting a record to a time-series database. As you have guessed by now, the field value, which is the actual data or the field set in general, as well as a timestamp, are essential components without which we can not save a point to the database. Although tag set is an optional component, it is a good practice to use, since it is indexed and would be useful in faster querying of data, as opposed to the field set, which is *not* indexed.
```

Comment:
```suggestion
The two components **tag set** and **field set** in the line protocol are sets, meaning a key-value pair. **Tag Key** and **Tag Value** are both stored in InfluxDB as strings. In contrast, only **Field Key** can be a string but the **Field Value** can contain strings, integers, floats, and Booleans too.  The field value is the actual data that we want to save to a database, and it is always accompanied by a timestamp, which makes sense since we are inserting a record to a time-series database.

As you have guessed by now, the field value (which is the actual data or the field set in general) as well as a timestamp, are essential components without which we cannot save a point to the database. Although tag set is an optional component, it is a good practice to use, since it is indexed and results in faster querying of data, as opposed to the field set, which is *not* indexed.
```